### PR TITLE
Coverage for facade

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- increase code coverage
  
## Proposed changes
- added unit tests on facade package

## Testing procedure
- tests only

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: facade/nodeFacade_test.go [hunk 1]
```diff
@@ -279,25 +335,43 @@ func TestNodeFacade_SetSyncer(t *testing.T) {
...
-	}
+	t.Run("should error", func(t *testing.T) {
+		t.Parallel()
 
-	arg := createMockArguments()
-	arg.Node = node
-	nf, _ := NewNodeFacade(arg)
+		arg := createMockArguments()
+		arg.Node = &mock.NodeStub{
+			GetAccountCalled: func(address string, _ api.AccountQueryOptions) (api.AccountResponse, api.BlockInfo, error) {
```
Comment (bogdan-rosianu) [inline facade/nodeFacade_test.go:349]: the address isn't used either

File: facade/nodeFacade_test.go [hunk 2]
```diff
@@ -1747,7 +1924,248 @@ func TestNodeFacade_InternalValidatorsInfo(t *testing.T) {
...
+	response, err := nf.GetInternalStartOfEpochMetaBlock(0, 0)
+	require.NoError(t, err)
+	require.Equal(t, providedResponse, response)
+}
+
+func TestNodeFacade_Close(t *testing.T) {
+	t.Parallel()
+
+	nf, _ := NewNodeFacade(createMockArguments())
+	require.NoError(t, nf.Close())
```
Comment (bogdan-rosianu) [inline facade/nodeFacade_test.go:2160]: it seems like the context created on the constructor isn't used anywhere. might remove it and the cancel function

Comment (sstanculeanu) [reply facade/nodeFacade_test.go:2160]: indeed, fixed

**Review by bogdan-rosianu — APPROVED** [review]

File: facade/nodeFacade.go [hunk 3]
```diff
@@ -68,8 +68,6 @@ type nodeFacade struct {
 	accountsState          state.AccountsAdapter
 	peerState              state.AccountsAdapter
 	blockchain             chainData.ChainHandler
-	ctx                    context.Context
```
Comment (iulianpascalau) [inline facade/nodeFacade.go:71]: :+1:
golang says: we should not store contexts

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
@@                            Log scanner                                  @@
##                            facade_tests                                 ##
================================================================================
- Known Warnings              10
- New Warnings                3
- Known Errors                1
- New Errors                  1
- Panics                      0
================================================================================
- block hash does not match                  11250
- wrong nonce in block                       4000
- miniblocks does not match                  0
- num miniblocks does not match              0
- miniblock hash does not match              0
- block bodies does not match                0
- receipts hash missmatch                    1
================================================================================
-  No jailed nodes on the testnet
================================================================================

**Pull request merged**